emloop_tensorflow/model.py

```diff
@@ -375,8 +375,12 @@ def _create_train_ops(self, dependencies: List[List[tf.Operation]], optimizer_co
         if optimizer_config is None:
             raise ValueError('Optimizer config was not specified although it is required for creating the train op. '
                              'Please specify the configuration in `model.optimizer`.')
+
+        optimizer_config_copy = optimizer_config.copy()
+        clip = optimizer_config_copy.pop("clip", None)
+
         grads_and_vars = []
-        optimizer = create_optimizer(optimizer_config)
+        optimizer = create_optimizer(optimizer_config_copy)
```

Comment:
Why is it necessary to pass a copy of the config here? Wouldn't it be sufficient to just extract the `clip` via `optimizer_config.get('clip', None)` ?

Reply:
Well, the error @bedapisl mentioned is quite natural since the `optimizer_config` is passed as-is to the selected `tf` optimizer constructor.

It would prefer a documented `Model` argument (`clip_gradient`?) so that we can avoid this...